src/service/service.py

```diff
@@ -86,26 +86,35 @@ async def info() -> ServiceMetadata:
     )
 
 
-def _parse_input(user_input: UserInput) -> tuple[dict[str, Any], UUID]:
+async def _parse_input(
+    user_input: UserInput, agent: CompiledStateGraph
+) -> tuple[dict[str, Any], UUID]:
     run_id = uuid4()
     thread_id = user_input.thread_id or str(uuid4())
+    messages = {"messages": [HumanMessage(content=user_input.message)]}
 
     configurable = {"thread_id": thread_id, "model": user_input.model}
-
     if user_input.agent_config:
         if overlap := configurable.keys() & user_input.agent_config.keys():
             raise HTTPException(
                 status_code=422,
                 detail=f"agent_config contains reserved keys: {overlap}",
             )
         configurable.update(user_input.agent_config)
+    config = RunnableConfig(configurable=configurable, run_id=run_id)
+
+    input = messages
+    # Human-in-the-loop: Retrieve the current state to check if an interruption has occurred.
+    snapshot = await agent.aget_state(config)
+    if snapshot.next:
+        resume_node_name = snapshot.next[0]
+        # Human-in-the-loop: Create a command to resume the node.
+        # https://langchain-ai.github.io/langgraph/concepts/human_in_the_loop/#validating-human-input
+        input = Command(goto=resume_node_name, update=messages)
```

Comment:
[NOTE]
Without this addition, it will unintentionally restart from the first node.